src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+                                        SelectableOptionMenuItemType.Header
+                                )
+                                .map((option) => option.data || option.text) ||
+                                []
+                        );
+                } else {
+                    setSelectedOption(existingOption);
+                    setSelectedKey(existingOption.key);
+                    if (onResourceChange)
+                        onResourceChange(
+                            existingOption.data || existingOption.text,
+                            options
+                                ?.filter(
+                                    (option) =>
+                                        option.itemType !==
+                                        SelectableOptionMenuItemType.Header
+                                )
+                                .map((option) => option.data || option.text) ||
+                                []
+                        );
+                }
+            }
+        },
+        [options]
+    );
+
+    const handleOnRemove = useCallback(
+        (option: IComboBoxOption) => {
+            const restOfOptions = options.filter((o) => o.key !== option.key);
+            if (restOfOptions.length) {
+                const lastOption = restOfOptions[restOfOptions.length - 1];
+                if (
+                    lastOption.itemType === SelectableOptionMenuItemType.Header
+                ) {
+                    // remove the '---' freeform options header if there is no option below
+                    restOfOptions.splice(-1);
+                }
+            }
+            setOptions(restOfOptions);
+
+            if (option.key === selectedOption?.key) {
+                setSelectedOption(null);
+                if (onResourceChange)
+                    onResourceChange(
+                        null,
+                        restOfOptions
+                            .filter(
+                                (option) =>
+                                    option.itemType !==
+                                    SelectableOptionMenuItemType.Header
+                            )
+                            .map((option) => option.data || option.text)
+                    );
+            } else {
+                if (onResourceChange)
+                    onResourceChange(
+                        selectedOption?.data || selectedOption?.text,
+                        restOfOptions
+                            ?.filter(
+                                (option) =>
+                                    option.itemType !==
+                                    SelectableOptionMenuItemType.Header
+                            )
+                            .map((option) => option.data || option.text)
+                    );
+            }
+        },
+        [options, selectedOption]
+    );
+
+    const handleOnRender = useCallback(
```

Comment:
naming nit: `onRenderOption`

Reply:
I prefer naming the handler functions to start with handle... where as naming the props start with on.... There was a stack post about it which I adopted